angular-workspace/projects/ni/nimble-angular/src/directives/combobox/nimble-combobox-control-value-accessor.directive.ts

```diff
@@ -119,20 +120,45 @@ export class NimbleComboboxControlValueAccessorDirective implements ControlValue
     /**
      * @internal
      */
-    public addOption(displayValue: string, modelValue: unknown): void {
-        this._optionMap.set(displayValue, modelValue);
+    public addOption(displayValue: string, modelValue: unknown, option: ListOption): void {
+        this._optionToModelValueMap.set(option, modelValue);
+        const options = this._displayTextToOptionsMap.get(displayValue);
+        if (options) {
+            const optionIndex = options.indexOf(option);
+            if (optionIndex < 0) {
+                options.push(option);
+            }
+        } else {
+            this._displayTextToOptionsMap.set(displayValue, [option]);
+        }
         this.updateDisplayValue();
     }
 
     /**
      * @internal
      */
-    public removeOption(displayValue: string): void {
-        this._optionMap.delete(displayValue);
+    public removeOption(displayValue: string, option: ListOption): void {
+        const options = this._displayTextToOptionsMap.get(displayValue);
+        if (options) {
+            if (options.length > 1) {
+                const removeIndex = options.indexOf(option);
+                if (removeIndex >= 0) {
+                    options.splice(removeIndex, 1);
+                }
+            } else {
+                this._displayTextToOptionsMap.delete(displayValue);
+            }
+        }
+        this._optionToModelValueMap.delete(option);
+    }
+
+    public updateOption(modelValue: unknown, option: ListOption): void {
+        this._optionToModelValueMap.set(option, modelValue);
+        this.updateDisplayValue();
     }
 
     public queueOptionUpdate(listOption: ListOption, modelValue: unknown): void {
```

Comment:
Can we align the signature with addOption and give it a more descriptive name, i.e.:

```ts
updateOptionModelValue(displayValue: string, modelValue: unknown, option: ListOption)
```

Concern is some of these methods access the `option.text` directly while others expect it to be passed in via `displayValue`, just trying to get it to be consistent.

Reply:
Yeah, I just got rid of that extra argument everywhere...it was unneeded.